### ISSUE TITLE
DbtTaskGroup - Kubernetes Execution with RenderConfig

### ISSUE THREAD

**Issue by harshag7dm** [astronomer/astronomer-cosmos#943]
When I use Kubernetes Execution in DbtTaskGroup and try to filter out the Models using tag, it doesn't work. There is no Kubernetes pod being run.
If I remove the RenderConfig, then all the models are run? 
Is this a bug or am I missing something?

**Comment by tatiana** [comment]
To try to reproduce this:
1. Run https://astronomer.github.io/astronomer-cosmos/getting_started/kubernetes.html#kubernetes
2. Modify https://github.com/astronomer/cosmos-example/blob/main/dags/jaffle_shop_kubernetes.py to have RenderConfig with a selector using tag (e.g. https://github.com/astronomer/cosmos-demo/blob/main/dags/filtering/customers_tag.py)
3. Confirm we can reproduce the issue
4. Fix this, if we can

**Comment by pankajastro** [comment]
This is a legitimate issue. I can replicate it using the steps outlined above. I’ll create a PR to address it.
<img width="1435" alt="Screenshot 2024-07-16 at 10 51 00 PM" src="https://github.com/user-attachments/assets/50e1effe-0556-4153-8ebd-5624cdd587c9">

**Comment by pankajastro** [comment]
I investigated further and found that when the manifest is unavailable and the execution mode isn't set to ExecutionMode.LOCAL, we're using the custom parser to load the dbt project. However, this custom parser currently does not support selectors.

https://github.com/astronomer/astronomer-cosmos/blob/2994dba1059cc914343a477e71564300b3fa0896/cosmos/dbt/graph.py#L323-L329

The workaround I tried 
- Use load_via_dbt_ls for other execution modes as well, although I'm uncertain about any potential side effects.
- Extract the tag in the custom parser method and assign it to the node. This is currently set to empty. https://github.com/astronomer/astronomer-cosmos/blob/2994dba1059cc914343a477e71564300b3fa0896/cosmos/dbt/graph.py#L557
